subsys/bluetooth/host/hci_core.c

```diff
@@ -1311,42 +1311,63 @@ static void enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
 #endif
 
 	if (evt->status) {
-		/*
-		 * If there was an error we are only interested in pending
-		 * connection. There is no need to check ID address as
-		 * only one connection can be in that state.
-		 *
-		 * Depending on error code address might not be valid anyway.
-		 */
-		conn = find_pending_connect(evt->role, NULL);
-		if (!conn) {
-			return;
-		}
-
-		conn->err = evt->status;
-
 		if (IS_ENABLED(CONFIG_BT_PERIPHERAL) &&
-		    conn->err == BT_HCI_ERR_ADV_TIMEOUT) {
+		    evt->status == BT_HCI_ERR_ADV_TIMEOUT) {
 			/*
 			 * Handle advertising timeout after high duty directed
 			 * advertising.
 			 */
 
+			/*
+			 * If there was an error we are only interested in
+			 * pending connection. There is no need to check ID
+			 * address as only one connection can be in that state.
+			 *
+			 * Depending on error code address might not be valid
+			 * anyway.
+			 */
+			conn = find_pending_connect(BT_HCI_ROLE_SLAVE, NULL);
+			if (!conn) {
+				BT_ERR("No pending slave connection");
+
+				return;
+			}
+
+			conn->err = evt->status;
+
 			atomic_clear_bit(bt_dev.flags, BT_DEV_ADVERTISING);
```

Comment:
I think we should take the oppurtunity to handle this as well:
```
		/* Clear advertising even if we are not able to add connection
		 * object to keep host in sync with controller state
		 */
		atomic_clear_bit(bt_dev.flags, BT_DEV_ADVERTISING);
```
Move above checking if no conn.